node/src/components/block_executor.rs

```diff
@@ -245,25 +286,38 @@ impl BlockExecutor {
     ) -> Effects<Event> {
         let next_deploy = match state.remaining_deploys.pop_front() {
             Some(deploy) => deploy,
-            None => {
-                // The state hash of the last execute-commit cycle is used as the block's post state
-                // hash.
-                let next_height = state.finalized_block.height() + 1;
-                let block = self.create_block(state.finalized_block, state.pre_state_hash);
-
-                let mut effects = effect_builder
-                    .announce_linear_chain_block(block, state.execution_results)
-                    .ignore();
-                // If the child is already finalized, start execution.
-                if let Some((finalized_block, deploys)) = self.exec_queue.remove(&next_height) {
-                    effects.extend(self.handle_get_deploys_result(
-                        effect_builder,
-                        finalized_block,
-                        deploys,
-                    ));
+            None => match state.finalized_block.era_end().as_ref() {
+                Some(era_end) => {
+                    let reward_items = era_end
+                        .rewards
+                        .iter()
+                        .map(|(&vid, &value)| RewardItem {
+                            validator_id: vid.into(),
+                            value,
+                        })
+                        .collect();
+                    let slash_items = era_end
+                        .equivocators
+                        .iter()
+                        .map(|&vid| SlashItem {
+                            validator_id: vid.into(),
+                        })
+                        .collect();
+                    let request = StepRequest {
+                        pre_state_hash: state.pre_state_hash.into(),
+                        protocol_version: ProtocolVersion::V1_0_0,
+                        reward_items,
+                        slash_items,
+                        run_auction: true,
+                    };
+                    return effect_builder
+                        .run_step(request)
+                        .event(|result| Event::RunStepResult { state, result });
                 }
-                return effects;
-            }
+                None => {
+                    return self.finalize_block_execution(effect_builder, state);
+                }
```

Comment:
I'd find that easier to read if we returned early in the `None` case. (Especially since this is already a `None` case inside another `None` case! :dizzy_face:)

Reply:
I'm not sure what you mean - we do return early here, don't we?

Reply:
It would already help me as a reader if the `None` case came first, because the `Some` case is pretty long.
Alternatively, by "returning early" I meant:
```rust
let era_end = match state.finalized_block.era_end().as_ref() {
    Some(era_end) => era_end,
    None => return self.finalize_block_execution(effect_builder, state),
};
let reward_items = era_end…
```
(But I realize that that wouldn't actually save as many levels of indentation as I thought.)

Reply:
(Anyway, not important. Feel free to merge as it is!)

Reply:
I like the alternative suggestion, so I applied it :+1: